http-gateway/pkg/server/topic_existence_checker.go

```diff
@@ -33,29 +45,57 @@ func NewAlwaysTrueTopicExistenceChecker() TopicExistenceChecker {
 // NewRiffTopicExistenceChecker configures a TopicExistenceChecker using the
 // provided Clientset.
 func NewRiffTopicExistenceChecker(clientSet *versioned.Clientset) TopicExistenceChecker {
-	return &riffTopicExistenceChecker{client: clientSet}
-}
+	riffInformerFactory := informers.NewSharedInformerFactory(clientSet, time.Second*30)
+	topicInformer := riffInformerFactory.Projectriff().V1alpha1().Topics()
+	knownTopics := make(map[string]ignoredValue)
+	mutex := &sync.Mutex{}
+
+	topicInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			mutex.Lock()
+			defer mutex.Unlock()
 
-func (tec *alwaysTrueTopicExistenceChecker) TopicExists(namespace string, topicName string) (bool, error) {
-	return true, nil
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			if err != nil {
+				log.Printf("AddFunc had an error for key '%s': %+v", err)
+			}
+
+			knownTopics[key] = ignoredValue{}
+			log.Printf("Added topic to internal map: %+v", key)
+		},
+
+		DeleteFunc: func(obj interface{}) {
+			mutex.Lock()
+			defer mutex.Unlock()
+
+			key, err := cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+			if err != nil {
+				log.Printf("DeleteFunc had an error for key '%s': %+v", err)
+			}
+
+			delete(knownTopics, key)
+			log.Printf("Removed topic from internal map: %+v", key)
+		},
+	})
+
+	done := make(chan struct{}) //TODO: ideally, this would be the same channel used by the gateway itself
+	go topicInformer.Informer().Run(done)
+
+	return &riffTopicExistenceChecker{topicInformer: topicInformer, knownTopics: knownTopics, mutex: mutex}
 }
 
-// TopicExists checks to see if Kubernetes is aware of a Riff Topic in a namespace.
-// If the topic exists, it returns (true, nil)
-// If the topic does not exist, it returns (false, nil)
-// If there is an unexpected error, it returns (false, err), where 'err' is the unexpected
-// error that was encountered.
-func (tec *riffTopicExistenceChecker) TopicExists(namespace string, topicName string) (bool, error) {
+func (tec *alwaysTrueTopicExistenceChecker) TopicExists(namespace string, topicName string) bool {
+	return true
+}
 
-	_, err := tec.client.ProjectriffV1alpha1().Topics(namespace).Get(topicName, v1.GetOptions{})
+// TopicExists checks to see if Kubernetes is aware of a riff Topic in a namespace.
+func (tec *riffTopicExistenceChecker) TopicExists(namespace string, topicName string) bool {
+	tec.mutex.Lock()
+	defer tec.mutex.Unlock()
 
-	if err != nil {
-		if k8serrors.IsNotFound(err) {
-			return false, nil
-		}
+	topicKey := fmt.Sprintf("%s/%s", namespace, topicName)
```

Comment:
This won't work if the key that came back from k8s was of the form `name` because the namespace is empty (whatever that means).